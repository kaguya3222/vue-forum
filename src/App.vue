<template>
  <div id="app">
    <nav-bar />
    <div class="container">
      <router-view />
    </div>
  </div>
</template>

<script>
import TheNavbar from "./components/TheNavbar";

export default {
  components: {
    "nav-bar": TheNavbar
  },
  beforeCreate() {
    this.$store.dispatch("users/fetchUser", {
      id: this.$store.state.users.authId
    });
  }
};
</script>

<style>
@import "./assets/css/style.css";
</style>
