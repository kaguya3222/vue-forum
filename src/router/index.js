import Vue from "vue";
import VueRouter from "vue-router";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "Home",
    component: () => import("@/pages/PageHome")
  },
  {
    path: "/me",
    name: "Profile",
    component: () => import("@/pages/PageProfile")
  },
  {
    path: "/me/edit",
    name: "ProfileEditor",
    component: () => import("@/pages/PageProfile"),
    props: { edit: true }
  },
  {
    path: "/category/:id",
    name: "Category",
    component: () => import("@/pages/PageCategory"),
    props: true
  },
  {
    path: "/forum/:id",
    name: "Forum",
    component: () => import("@/pages/PageForum"),
    props: true
  },
  {
    path: "/thread/create/:forumId",
    name: "ThreadCreate",
    component: () => import("@/pages/PageThreadCreate"),
    props: true
  },
  {
    path: "/thread/:threadId",
    name: "ThreadShow",
    component: () => import("@/pages/PageThreadShow"),
    props: true
  },
  {
    path: "*",
    name: "NotFound",
    component: () => import("@/pages/PageNotFound")
  }
];

const router = new VueRouter({
  mode: "history",
  routes
});

export default router;
