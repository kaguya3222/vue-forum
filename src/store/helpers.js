import Vue from "vue";

const makeAppendChildToParentMutation = ({
  parents,
  child,
  parentModuleName
}) => (state, { childId, parentId, rootState }) => {
  const resource = rootState[parentModuleName][parents][parentId];
  if (!resource[child]) {
    Vue.set(resource, child, {});
  }
  Vue.set(resource[child], childId, childId);
};

export { makeAppendChildToParentMutation };
