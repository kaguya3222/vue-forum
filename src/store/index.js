import Vue from "vue";
import Vuex from "vuex";
import sourceData from "@/data.json";
import forumPosts from "./modules/posts/store";
import forumUsers from "./modules/users/store";

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    ...sourceData
  },
  getters: {},
  mutations: {},
  actions: {},
  modules: {
    forumPosts,
    forumUsers
  }
});
