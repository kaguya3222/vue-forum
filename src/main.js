import Vue from "vue";
import App from "./App.vue";
import "./registerServiceWorker";
import router from "./router";
import store from "./store";
import AppDate from "./components/AppDate";

Vue.config.productionTip = false;
Vue.component("app-date", AppDate);

new Vue({
  router,
  store,
  render: h => h(App)
}).$mount("#app");
