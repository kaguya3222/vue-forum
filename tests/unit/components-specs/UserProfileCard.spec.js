import { mount } from "@vue/test-utils";
import UserProfileCard from "../../../src/components/UserProfileCard";
import mockedSourceData from "../mocks/mockedSourceData";
import { countObjectProperties } from "../../../src/helpers";

describe("UserProfileCard", () => {
  const user = Object.values(mockedSourceData.users)[0];
  const wrapper = mount(UserProfileCard, {
    propsData: {
      user,
      userPostsCount: countObjectProperties(user.posts),
      userThreadsCount: countObjectProperties(user.threads)
    },
    stubs: ["router-link"]
  });
  test("Correctly shows user profile card data", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
