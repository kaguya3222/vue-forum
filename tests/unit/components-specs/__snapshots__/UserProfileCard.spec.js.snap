// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileCard Correctly shows user profile card data 1`] = `
<div class="col-3 push-top">
  <div class="profile-card">
    <p class="text-center"><img src="https://avatars3.githubusercontent.com/u/2327556?v=4&amp;s=460" alt="" class="avatar-xlarge"></p>
    <h1 class="title">chrisvfritz</h1>
    <p class="text-lead">Chris Fritz</p>
    <p class="text-justify"><span>No bio specified.</span></p> <span class="online">chrisvfritz is online</span>
    <div class="stats"><span>0 posts</span> <span>0 threads</span></div>
    <hr>
    <!---->
  </div>
  <p class="text-xsmall text-faded text-center">
    Member since june 2003, last visited 4 hours ago
  </p>
  <div class="text-center">
    <hr>
    <router-link-stub to="[object Object]" class="btn-green btn-small">Edit profile</router-link-stub>
  </div>
</div>
`;
